src/Tasks/AssemblyDependency/BadImageReferenceException.cs

```diff
@@ -29,5 +29,10 @@ private BadImageReferenceException(SerializationInfo info, StreamingContext cont
             : base(info, context)
         {
         }
+
+        /// <summary>
+        /// Gets a message that describes the exception.
+        /// </summary>
+        public override string Message => (InnerException == null) ? Message : $"{Message} {InnerException.Message}";
```

Comment:
Will this recurse into itself infinitely?

Also should we display the full chain if there is more than one level of nesting?

Something like this perhaps?

```csharp
    private static string GetExceptionMessage(Exception ex)
    {
        string message = null;

        while (ex != null)
        {
            if (message == null)
            {
                message = ex.Message;
            }
            else
            {
                message += $"{message}: {ex.Message}";
            }

            ex = ex.InnerException;
        }

        return message;
    }
```

Reply:
sWe only expect certain inner exception types here, some basic corelib ones that shouldn't have their own inner exceptions as far as I know. I think we'd only want to add such code if we knew there was a scenario. 

If we wanted to do this speculatively, we'd have to find every single place in the codebase we log an exception message and feed it to your helper code above, since corelib exceptions in general do not do it themselves. In fact I always figured the "message concatenation" was an MSBuild invention (I think I filed a bug long long ago for a particular BCL exception message that did not end in a period..). 

I don't think that churn is necessary.

AggregateException does it of course, but I think the way it stringifies them is ugly. I've tried in the past to change it.

Reply:
> Will this recurse into itself infinitely?

Oops, fixed. Not sure why tests passed for me.

Reply:
OK sounds good to me